script.js

```diff
@@ -0,0 +1,185 @@
+    removed from the errorHolder element and the class list of visible is added. If there
+    are no errors in the errors array the class of hidden is removed from the convertFromHolder
+    element and the class if visible is added.
+    */
+        function onChange(){
+            const errors = [];
+            const errorHolder = document.getElementById('errorHolder');
+            const errorList = document.getElementById('errorList');
+            userInput = parseInt(userInput);
+            if (userInput.id == ""){
+            errors.push(alert('Error: You must enter something'));
+            }
+                else if (isNaN(userInput)){
+                errors.push(alert('Error: You must enter a number'));
+            }
+
+            for (error of errors){
+                const errorText = document.createTextNode(error);
+                errorList.createElement('li');
+                errorList.innerHTML = errorText;
+                errorHolder.appendChild(errorList);
+            }
+
+            if (error.length > 0){
+                errorHolder.classList.remove('hidden');
+                errorHolder.classList.add('visible');
+            }
+                else {
+                convertFromHolder.classList.remove('hidden');
+                convertFromHolder.classList.add('visible');
+                }
+                
+        }
+/*These statements assign the elements with the ids of convertFrom and
+option to the variables with the same name.
+*/
+    const convertFrom = document.getElementsById('convertFrom');
+    const option = getElementsByTagName('option');
+    /*This is the event listenter for the select event on the option
+    element it has a callback function onSelect.
+    */
+    option.addEventListener('select', onSelect);
+    /*This is the callback function onSelect it checks the value of the option
+    element selected and creates option elements based on what has been selected
+    in the option elements in the convertFrom field assigns them a value and appends
+    them to the convertTo element.
+    */
+        function onSelect(){
+            const convertToHolder = document.getElementById('convertToHolder');
+            const convertTo = document.getElementById('convertTo');
+                if(!convertFrom.option.id == ''){
+            convertToHolder.classList.remove('hidden');
+            convertToHolder.classList.add('visible');
+                } 
+                else if(convertFrom.option.id == 'meters'){
+                    let option1 = document.createElement('option');
+                    let attribute1 = document.createAttribute('id');
+                    attribute1.value = 'feet'; 
+                    option1.setAttribute(attribute1);              
+                    let attribute2 = document.createAttribute('id');
+                    attribute2.value = 'miles';
+                    option1.setAttribute(attribute2);
+                    document.appendChild(convertTo);
```

Comment:
then append both options to convertTo - rather than what you are doing here - which is appending cocnvertTo to the document:

convertTo.appendChild(option1);
convertTo.appendChild(option2);

This should help you clean up all your elses here.